_pages/dev/features/product-multi-dimensional.md

```diff
@@ -0,0 +1,196 @@
+
+- product details pages
+- cart items (for example, see the `CartItemComponent`)
+- account pages, such as the order page, the order history, wish list, cancellations and returns, and so on.
+
+**Note:** The `$contentCV` variable that is used in the following ImpEx examples, and which stores information about the content catalog, is defined as follows:
+
+```text
+$contentCatalog=powertools-spaContentCatalog
+$contentCV=catalogVersion(CatalogVersion.catalog(Catalog.id[default=$contentCatalog]),CatalogVersion.version[default=Staged])[default=$contentCatalog:Staged]
+```
+
+1. Create a `ProductMultiDimensionalSelector` CMS flex component with the following ImpEx:
+
+   ```text
+   INSERT_UPDATE CMSFlexComponent;$contentCV[unique=true];uid[unique=true];name;flexType;&componentRef
+   ;;ProductMultiDimensionalSelector;Product Multi-Dimensional Selector Component;ProductMultiDimensionalSelectorComponent;ProductMultiDimensionalSelector
+   ```
+
+2. Add the newly created CMS flex component to the `ProductSummarySlot` with the following ImpEx:
+
+```text
+INSERT_UPDATE ContentSlot;$contentCV[unique=true];uid[unique=true];name;cmsComponents(uid, $contentCV)
+;;ProductSummarySlot;Summary for product details;ProductImagesComponent,ProductIntroComponent,QualtricsEmbeddedFeedbackComponent,ProductSummaryComponent,ProductMultiDimensionalSelector,AddToCart,ConfigureProductComponent,AddToWishListComponent,StockNotificationComponent
+```
+
+## Images
+
+Variant categories can use images for selecting the categories by setting the `hasImage` attribute to true. This can be configured in Backoffice, or through ImpEx. The `hasImage` attribute acts as a toggle that informs the storefront to display images for variant options. However, images are only shown if each variant option within the category has an image in the correct format. The default image format for multi-dimensional products is `STYLE_SWATCH`, which is 30 W x 30 H.
+
+## List Item Details Component
+
+This functionality replaces `ProductListOutlets.ITEM_DETAILS` with the new `ProductMultiDimensionalListItemDetailsComponent`. Its primary function is to display the price range for multi-dimensional products within the grid, or to list item components on the product listing page (PLP).
+
+## Multi-Dimensional Product Guard
+
+When a user encounters a product with variant categories, the `ProductMultiDimensionalSelectorGuard` allows Spartacus to redirect the user under the following conditions:
+
+- If the selected variant cannot be purchased (for example, it is the base product, or the variant is out of stock), the `ProductMultiDimensionalSelectorGuard` identifies the closest available variant with stock and redirects the user to its product details page.
+- If the selected variant is purchasable (because it is not the base product) but all variants are out of stock, the guard redirects to the first variant product.
+
```

Comment:
I don't quite understand what is meant by "the guard redirects to the first variant product".

If all variants are out of stock... Is the user redirected to the first of the variants (which is out of stock) ? If so, what determines which variant is the first variant? Or does it mean the user is redirected to another product.... ? 😬 😄

Reply:
On the PLP, we display the base product, such as "Camera W-80." This base product is not available for purchase on its own, but it has purchasable variants like "Camera W-80 Black" and "Camera W-80 Blue." When a user clicks on the base product in the PLP, the guard searches for a valid variant. If all variant products are out of stock, the system redirects the user to the first variant of the base product, which in this case would be "Camera W-80 Black."

Reply:
Perfect! Thank you for clarifying 🙂